.github/workflows/syntax_check_cloud.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will check for exceptions to raise properly and do not generate syntax errors.
+# Also old code style using parenthesis in class definitions will be checked
+
+name: Syntax checking
+
+on:
+  [push, pull_request]
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04, ubuntu-22.04]
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Check exceptions
+      # Not related to tests, but to QA in general: Exceptions usually take only 1 argument
+      # So a comma is probably a sign of syntax error and should be replaced by a %
+      run: |
+        grep -IrE --colour 'raise .+".*%.*",' --include=*.py ./src/odemis ./scripts/ ./plugins/
+        if [[ $? -eq 0 ]]; then
+           echo "Found the above syntax errors in exceptions, this is a sign that a comma should be replaced by a %."        
+           exit 1
+        else
+           exit 0
```

Comment:
No, that will not work, because the script automatically stops when `grep` returns 1. (That's the `-e` flag of bash, which stops the script as soon as a line reports an error. Hence you need the `||` just after the `grep`.

Reply:
Alright, cool thanks for you input, I applied you suggestion and it seems to work nicely